components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImplTest.java

```diff
@@ -0,0 +1,336 @@
+                buildMockBasicAuthentication("updatingadmin", "updatingadmin"));
+        Action result = serviceImpl.updateAction(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(),
+                action.getId(), updatingAction, tenantDomain);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(updatingAction.getName(), result.getName());
+        Assert.assertEquals(updatingAction.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(updatingAction.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(
+                updatingAction.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType()
+                           );
+        action = result;
+    }
+
+    @Test(priority = 6)
+    public void testDeactivateAction() throws ActionMgtException {
+
+        Assert.assertEquals(Action.Status.ACTIVE, action.getStatus());
+        Action deactivatedAction = serviceImpl.deactivateAction(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(Action.Status.INACTIVE, deactivatedAction.getStatus());
+    }
+
+    @Test(priority = 7)
+    public void testActivateAction() throws ActionMgtException {
+
+        Action result = serviceImpl.activateAction(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(Action.Status.ACTIVE, result.getStatus());
+    }
+
+    @Test(priority = 8)
+    public void testGetActionByActionId() throws ActionMgtException {
+
+        Action result = serviceImpl.getActionByActionId(action.getType().getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 9)
+    public void testGetActionsCountPerType() throws ActionMgtException {
+
+        Map<String, Integer> actionMap = serviceImpl.getActionsCountPerType(tenantDomain);
+        for (Map.Entry<String, Integer> entry: actionMap.entrySet()) {
+            Assert.assertEquals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType(), entry.getKey());
+            Assert.assertEquals(1, entry.getValue().intValue());
```

Comment:
Shouldn't we need to assert the count values for other action types

Reply:
Updated in